cfgov/unprocessed/js/modules/footer-button.js

```diff
@@ -1,26 +1,64 @@
 /* ==========================================================================
    Footer Button: Scroll to Top
+
+   Code copied from the following with minimal modifications :
+
+   - http://stackoverflow.com/questions/21474678/
+     scrolltop-animation-without-jquery
    ========================================================================== */
+
 'use strict';
 
-var $ = require( 'jquery' );
+// Required modules.
+var dataHook = require( './util/data-hook' );
 
-// TODO: Refactor this file to remove jquery dependency.
-//       http://stackoverflow.com/questions/21474678/
-//       scrolltop-animation-without-jquery
 /**
  * Set up event handler for button to scroll to top of page.
  */
 function init() {
-  var duration = 300;
+  var scrollDuration = 300;
+  var behaviorElement = dataHook.get( 'return-to-top' );
 
-  // Disable Google Tag Manager tracking on this link.
-  $( '.js-return-to-top' ).attr( 'data-gtm_ignore', 'true' );
+  if ( behaviorElement === null ||
+       'requestAnimationFrame' in window === false ) {
+    return;
+  }
 
-  $( '.js-return-to-top' ).click( function( event ) {
+  // Disable Google Tag Manager tracking on this link.
+  behaviorElement.setAttribute( 'data-gtm_ignore', 'true' );
```

Comment:
The rationale for keeping it in the JS elsewhere is that (a) GTM lives only in JavaScript and (b) in some cases if the default link behavior is being prevented by a particular script, if that script were to be disabled/removed the links should _not_ continue ignoring GTM. In that case it makes sense to set `data-gtm_ignore` where `event.preventDefault()` is set.